### ISSUE TITLE
Implement ByteArray in Inko

### ISSUE THREAD

**Issue by yorickpeterse** [inko-lang/inko#594]
### Description

Extracting this from https://github.com/inko-lang/inko/issues/349:

`Array` is in the process of being rewritten to use only Inko code, instead of relying on a partial runtime library implementation. For `ByteArray` this is more tricky, as various runtime/Rust functions we use require a `Vec<u8>` as their input (e.g. when reading a file), meaning we can't just pass a pointer/slice to some bytes. https://github.com/inko-lang/inko/issues/349#issuecomment-1622437437 contains some extra details.

This will likely be a process involving many small steps, such as moving IO reading/writing logic into the standard library, though I'm not sure yet what the exact steps are at this time.

In an ideal world, we'd introduce a `Byte` type and just replace `ByteArray` with `Array[Byte]`. I'm not sure about this just yet though: while a `Byte` type has its use (certainly more so than e.g. `Int16` or `Int32`), it results in duplication of methods/logic between `Byte` and `Int`. We can make the `Byte` type a more bare-bones `Int`, requiring explicit casting, but that too is annoying.

In short: this will require more data/thinking before we start changing any code.

**Comment by yorickpeterse** [comment]
With the file and socket types now being implemented in Inko, we should be able to also implement `ByteArray` in Inko entirely. There are still a few places where we reference `ByteArray` in the runtime library, but those references can be replaced by a combination of a raw pointer and a size argument.
